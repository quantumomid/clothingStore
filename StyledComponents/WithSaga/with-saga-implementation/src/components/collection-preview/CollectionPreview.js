import { withRouter } from "react-router-dom";
import CollectionItem from '../collection-item/CollectionItem'
import { CollectionPreviewContainer, PreviewContainer, TitleContainer } from './collectionPreviewStyles'

const CollectionPreview = ({ title, items, history, match, routeName }) => (
        <CollectionPreviewContainer>
            <TitleContainer onClick={() => history.push(`${match.path}/${routeName}`)}>{title.toUpperCase()}</TitleContainer>
            <PreviewContainer>
                {items.filter((item, index) => index <4 ).map((item) => {
                    return <CollectionItem key={item.id} item={item} />
                })}
            </PreviewContainer>
        </CollectionPreviewContainer>
    )

export default withRouter(CollectionPreview);