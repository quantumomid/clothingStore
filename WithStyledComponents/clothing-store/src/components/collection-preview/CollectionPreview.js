import './CollectionPreview.scss'
import CollectionItem from '../collection-item/CollectionItem'

export default function CollectionPreview({ title, items }){

    return(
        <div className="collection-preview">
            <h1>{title.toUpperCase()}</h1>
            <div className="preview">
                {items.filter((item, index) => index <4 ).map((item) => {
                    return <CollectionItem key={item.id} item={item} />
                })}
            </div>
        </div>
    )
}