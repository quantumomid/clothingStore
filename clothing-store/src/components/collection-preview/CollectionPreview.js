import './CollectionPreview.scss'
import CollectionItem from '../collection-item/CollectionItem'

export default function CollectionPreview({ title, items }){

    return(
        <div className="collection-preview">
            <h1>{title.toUpperCase()}</h1>
            <div className="preview">
                {items.filter((item, index) => index <4 ).map(({ id, ...otherItemProps }) => {
                    return <CollectionItem key={id} {...otherItemProps} />
                })}
            </div>
        </div>
    )
}