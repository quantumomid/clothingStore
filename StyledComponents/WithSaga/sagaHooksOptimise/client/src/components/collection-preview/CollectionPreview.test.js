import React from 'react';
import { shallow } from 'enzyme';
import CollectionPreview from "./CollectionPreview";

describe('CollectionPreview component', () => {
  let wrapper;
  let mockMatch;
  let mockHistory;
  const mockRouteName = 'hats';

    beforeEach(() => {
    
        mockMatch = {
            path: '/shop'
        }
        
        mockHistory = {
            push: jest.fn()
        };

        const mockProps = {
            match: mockMatch,
            history: mockHistory,
            routeName: mockRouteName,
            title: 'hats',
            items: []
        };

        wrapper = shallow(<CollectionPreview {...mockProps} />);

    })

    afterEach(() => {
        jest.restoreAllMocks();
    });

    it('Should render CollectionPreview component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    // // Some issue with finding the title container below
    // it('Should call history.push with the right string when TitleContainer clicked', () => {
    //     wrapper.find("TitleContainer").simulate('click');
    
    //     expect(mockHistory.push).toHaveBeenCalledWith(
    //       `${mockMatch.path}/${mockRouteName}`
    //     );
    // });
})