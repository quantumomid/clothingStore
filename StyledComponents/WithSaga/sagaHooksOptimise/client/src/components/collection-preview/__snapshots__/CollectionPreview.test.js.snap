// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPreview component Should render CollectionPreview component 1`] = `ShallowWrapper {}`;
